--- origsrc/totem-2.24.3/configure.in	2008-10-26 09:40:05.000000000 -0500
+++ src/totem-2.24.3/configure.in	2008-12-15 01:50:52.587625000 -0600
@@ -830,6 +830,22 @@
 AC_SUBST([AM_CXXFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
+dnl ================================================================
+dnl Win32 plugin support
+dnl ================================================================
+case "$host_os" in
+	cygwin*|mingw*)
+		platform_win32=yes
+		win32_plugin_ldflags="-no-undefined -Wl,\$(top_builddir)/src/libtotem.a"
+		;;
+	*)
+		platform_win32=no
+		win32_plugin_ldflags=
+		;;
+esac
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(win32_plugin_ldflags)
+
 AC_OUTPUT([
 Makefile
 totem.spec
--- origsrc/totem-2.24.3/lib/Makefile.am	2008-09-01 05:32:56.000000000 -0500
+++ src/totem-2.24.3/lib/Makefile.am	2008-12-15 01:50:52.603250000 -0600
@@ -27,6 +27,7 @@
 	$(AM_LDFLAGS)
 
 libtotemscrsaver_la_LIBADD =				\
+	$(GTK_LIBS)						\
 	$(DBUS_LIBS)					\
 	$(XTEST_LIBS)
 
--- origsrc/totem-2.24.3/src/Makefile.am	2008-09-01 05:32:55.000000000 -0500
+++ src/totem-2.24.3/src/Makefile.am	2008-12-15 13:04:23.353250000 -0600
@@ -1,4 +1,4 @@
-SUBDIRS = plugins backend
+SUBDIRS = backend . plugins
 
 bin_PROGRAMS = totem totem-video-thumbnailer totem-video-indexer totem-audio-preview
 libexec_PROGRAMS =
@@ -41,6 +41,11 @@
 libbaconmessageconnection_la_LDFLAGS = \
 	$(AM_LDFLAGS)
 
+plugins/libtotemmodule.la:
+	cd $(@D) ; $(MAKE) $(@F)
+
+plugins/properties/libbaconvideowidgetproperties.la:
+	cd $(@D) ; $(MAKE) $(@F)
 
 EGGDIR=$(srcdir)/../../libegg/libegg/recent-files/
 BACONDIR=$(srcdir)/../../libbacon/src
@@ -139,11 +144,16 @@
 totem_LDFLAGS = \
 	$(AM_LDFLAGS)
 
+if PLATFORM_WIN32
+totem_LDFLAGS += \
+	-Wl,--export-all-symbols -Wl,--out-implib,libtotem.a
+endif
+
 totem_LDADD =						\
-	backend/libbaconvideowidget.la			\
 	plugins/libtotemmodule.la			\
 	libbaconmessageconnection.la			\
 	libtotem_player.la				\
+	backend/libbaconvideowidget.la			\
 	$(EXTRA_GNOME_LIBS)				\
 	$(XVIDMODE_LIBS)				\
 	$(NVTV_LIBS)					\
@@ -257,9 +267,9 @@
 	$(AM_LDFLAGS)
 
 test_properties_page_LDADD = 					\
-	backend/libbaconvideowidget.la				\
 	plugins/properties/libbaconvideowidgetproperties.la	\
 	libtotem_player.la					\
+	backend/libbaconvideowidget.la				\
 	$(EXTRA_GNOME_LIBS)					\
 	$(NAUTILUS_LIBS)					\
 	$(NVTV_LIBS)						\
--- origsrc/totem-2.24.3/src/plugins/galago/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/galago/Makefile.am	2008-12-15 01:50:52.665750000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/galago
 plugin_LTLIBRARIES = libtgp.la
--- origsrc/totem-2.24.3/src/plugins/gromit/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/gromit/Makefile.am	2008-12-15 01:50:52.681375000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/gromit
 plugin_LTLIBRARIES = libgromit.la
@@ -22,7 +22,7 @@
 
 libgromit_la_SOURCES = totem-gromit.c
 libgromit_la_LDFLAGS = $(modules_flags)
-libgromit_la_LIBADD = 
+libgromit_la_LIBADD = $(DBUS_LIBS)
 libgromit_la_CPPFLAGS = $(common_defines)
 
 libgromit_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/lirc/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/lirc/Makefile.am	2008-12-15 01:50:52.712625000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/lirc
 plugin_LTLIBRARIES = liblirc.la
--- origsrc/totem-2.24.3/src/plugins/media-player-keys/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/media-player-keys/Makefile.am	2008-12-15 01:50:52.712625000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/media-player-keys
 plugin_LTLIBRARIES = libmedia_player_keys.la
--- origsrc/totem-2.24.3/src/plugins/ontop/Makefile.am	2008-09-01 05:32:53.000000000 -0500
+++ src/totem-2.24.3/src/plugins/ontop/Makefile.am	2008-12-15 13:26:54.931375000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/ontop
 plugin_LTLIBRARIES = libontop.la
@@ -22,7 +22,7 @@
 
 libontop_la_SOURCES = totem-ontop.c
 libontop_la_LDFLAGS = $(modules_flags)
-libontop_la_LIBADD = $(top_builddir)/lib/libtotemscrsaver.la
+libontop_la_LIBADD = $(top_builddir)/lib/libtotemscrsaver.la $(top_builddir)/src/backend/libbaconvideowidget.la
 libontop_la_CPPFLAGS = $(common_defines)
 
 libontop_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/properties/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/properties/Makefile.am	2008-12-15 13:29:46.072000000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/properties
 plugin_LTLIBRARIES = libmovie-properties.la
@@ -24,7 +24,7 @@
 
 libmovie_properties_la_SOURCES = totem-movie-properties.c
 libmovie_properties_la_LDFLAGS = $(modules_flags)
-libmovie_properties_la_LIBADD = libbaconvideowidgetproperties.la
+libmovie_properties_la_LIBADD = libbaconvideowidgetproperties.la $(top_builddir)/src/backend/libbaconvideowidget.la $(GTK_LIBS)
 libmovie_properties_la_CPPFLAGS = $(common_defines)
 
 libmovie_properties_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/publish/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/publish/Makefile.am	2008-12-15 01:50:52.759500000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/publish
 plugin_LTLIBRARIES = libpublish.la
--- origsrc/totem-2.24.3/src/plugins/sample-vala/Makefile.am	2008-09-01 05:32:53.000000000 -0500
+++ src/totem-2.24.3/src/plugins/sample-vala/Makefile.am	2008-12-15 01:50:52.775125000 -0600
@@ -7,7 +7,7 @@
 %.totem-plugin: %.totem-plugin.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -d -u -c $(top_builddir)/po/.intltool-merge-cache
 
 if ENABLE_VALA
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugin_LTLIBRARIES = libsample-vala.la
 # override to _not_ install the test plugins
@@ -29,6 +29,7 @@
 
 nodist_libsample_vala_la_SOURCES = $(BUILT_SOURCES)
 libsample_vala_la_LDFLAGS = $(modules_flags)
+libsample_vala_la_LIBADD = $(VALA_LIBS)
 libsample_vala_la_CPPFLAGS = $(common_defines)
 
 libsample_vala_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/screensaver/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/screensaver/Makefile.am	2008-12-15 13:22:21.290750000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/screensaver
 plugin_LTLIBRARIES = libscreensaver.la
@@ -22,7 +22,7 @@
 
 libscreensaver_la_SOURCES = totem-screensaver.c
 libscreensaver_la_LDFLAGS = $(modules_flags)
-libscreensaver_la_LIBADD = $(top_builddir)/lib/libtotemscrsaver.la
+libscreensaver_la_LIBADD = $(top_builddir)/lib/libtotemscrsaver.la $(top_builddir)/src/backend/libbaconvideowidget.la
 libscreensaver_la_CPPFLAGS = $(common_defines)
 
 libscreensaver_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/sidebar-test/Makefile.am	2008-09-01 05:32:53.000000000 -0500
+++ src/totem-2.24.3/src/plugins/sidebar-test/Makefile.am	2008-12-15 01:50:52.822000000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/sidebar-test
 plugin_LTLIBRARIES = libsidebar-test.la
@@ -24,6 +24,7 @@
 
 libsidebar_test_la_SOURCES = totem-sidebar-test.c
 libsidebar_test_la_LDFLAGS = $(modules_flags)
+libsidebar_test_la_LIBADD = $(GTK_LIBS)
 libsidebar_test_la_CPPFLAGS = $(common_defines)
 
 libsidebar_test_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/skipto/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/skipto/Makefile.am	2008-12-15 13:36:00.712625000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/skipto
 uidir = $(plugindir)
@@ -28,6 +28,7 @@
 	totem-skipto.c		\
 	totem-skipto.h
 libskipto_la_LDFLAGS = $(modules_flags)
+libskipto_la_LIBADD = $(top_builddir)/src/backend/libbaconvideowidget.la $(GTK_LIBS)
 libskipto_la_CPPFLAGS = \
 	$(common_defines)		\
 	-I$(top_srcdir)/src/backend	\
--- origsrc/totem-2.24.3/src/plugins/thumbnail/Makefile.am	2008-09-01 05:32:53.000000000 -0500
+++ src/totem-2.24.3/src/plugins/thumbnail/Makefile.am	2008-12-15 01:50:52.853250000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/thumbnail
 plugin_LTLIBRARIES = libthumbnail.la
@@ -22,7 +22,7 @@
 
 libthumbnail_la_SOURCES = totem-thumbnail.c
 libthumbnail_la_LDFLAGS = $(modules_flags)
-libthumbnail_la_LIBADD = 
+libthumbnail_la_LIBADD = $(GTK_LIBS)
 libthumbnail_la_CPPFLAGS = $(common_defines)
 
 libthumbnail_la_CFLAGS = \
--- origsrc/totem-2.24.3/src/plugins/tracker/Makefile.am	2008-09-01 05:32:54.000000000 -0500
+++ src/totem-2.24.3/src/plugins/tracker/Makefile.am	2008-12-15 01:50:52.884500000 -0600
@@ -1,4 +1,4 @@
-modules_flags = -export_dynamic -avoid-version -module
+modules_flags = -export_dynamic -avoid-version -module $(win32_plugin_ldflags)
 
 plugindir = $(PLUGINDIR)/tracker
 plugin_LTLIBRARIES = libtracker.la
