--- origsrc/totem-2.28.2/browser-plugin/Makefile.am	2009-09-29 09:47:24.000000000 -0500
+++ src/totem-2.28.2/browser-plugin/Makefile.am	2009-11-09 03:21:21.203728800 -0600
@@ -191,6 +191,7 @@ libtotem_gmp_plugin_la_LDFLAGS = \
 	-avoid-version	\
 	-export-symbols $(srcdir)/plugin.symbols \
 	-module		\
+	-no-undefined	\
 	$(AM_LDFLAGS)
 
 endif
@@ -251,6 +252,7 @@ libtotem_complex_plugin_la_LDFLAGS = \
 	-avoid-version	\
 	-export-symbols $(srcdir)/plugin.symbols \
 	-module		\
+	-no-undefined	\
 	$(AM_LDFLAGS)
 
 endif
@@ -311,6 +313,7 @@ libtotem_narrowspace_plugin_la_LDFLAGS =
 	-avoid-version	\
 	-export-symbols $(srcdir)/plugin.symbols \
 	-module		\
+	-no-undefined	\
 	$(AM_LDFLAGS)
 
 endif
@@ -371,6 +374,7 @@ libtotem_mully_plugin_la_LDFLAGS = \
 	-avoid-version	\
 	-export-symbols $(srcdir)/plugin.symbols \
 	-module		\
+	-no-undefined	\
 	$(AM_LDFLAGS)
 
 endif
@@ -441,6 +445,7 @@ libtotem_cone_plugin_la_LDFLAGS = \
 	-avoid-version	\
 	-export-symbols $(srcdir)/plugin.symbols \
 	-module		\
+	-no-undefined	\
 	$(AM_LDFLAGS)
 
 endif
--- origsrc/totem-2.28.2/browser-plugin/totemPluginGlue.cpp	2009-10-21 09:17:00.000000000 -0500
+++ src/totem-2.28.2/browser-plugin/totemPluginGlue.cpp	2009-11-09 03:09:20.117471400 -0600
@@ -430,7 +430,11 @@ NP_Initialize (NPNetscapeFuncs *aMozilla
 	/* we want to open libdbus-glib-1.so.2 in such a way
 	 * in such a way that it becomes permanentely resident */
 	void *handle;
+#ifdef __CYGWIN__
+	handle = dlopen ("cygdbus-glib-1-2.dll", RTLD_NOW);
+#else
 	handle = dlopen ("libdbus-glib-1.so.2", RTLD_NOW | RTLD_NODELETE);
+#endif
 	if (!handle) {
 		fprintf (stderr, "%s\n", dlerror()); 
 		return NPERR_MODULE_LOAD_FAILED_ERROR;
