--- origsrc/totem-3.0.1/browser-plugin/Makefile.am	2011-04-26 05:22:22.000000000 -0500
+++ src/totem-3.0.1/browser-plugin/Makefile.am	2011-05-25 04:37:53.040347900 -0500
@@ -49,6 +49,7 @@ plugin_ldflags = \
 	-avoid-version					\
 	-export-symbols $(srcdir)/plugin.symbols	\
 	-module						\
+	-no-undefined					\
 	$(AM_LDFLAGS)
 
 # Marshaller
--- origsrc/totem-3.0.1/browser-plugin/totemPluginGlue.cpp	2010-10-18 08:19:58.000000000 -0500
+++ src/totem-3.0.1/browser-plugin/totemPluginGlue.cpp	2011-05-25 04:39:00.561209900 -0500
@@ -434,7 +434,11 @@ NP_Initialize (NPNetscapeFuncs *aMozilla
 	/* we want to open libdbus-glib-1.so.2 in such a way
 	 * in such a way that it becomes permanentely resident */
 	void *handle;
+#ifdef __CYGWIN__
+	handle = dlopen ("cygdbus-glib-1-2.dll", RTLD_NOW);
+#else
 	handle = dlopen ("libdbus-glib-1.so.2", RTLD_NOW | RTLD_NODELETE);
+#endif
 	if (!handle) {
 		fprintf (stderr, "%s\n", dlerror()); 
 		return NPERR_MODULE_LOAD_FAILED_ERROR;
--- origsrc/totem-3.0.1/src/Makefile.am	2011-04-26 05:22:22.000000000 -0500
+++ src/totem-3.0.1/src/Makefile.am	2011-05-25 04:41:47.992786400 -0500
@@ -106,7 +106,8 @@ libtotem_la_CFLAGS = \
 	$(AM_CFLAGS)
 
 libtotem_la_LDFLAGS = \
-	-export_dynamic	\
+	-export-dynamic	\
+	-no-undefined \
 	$(AM_LDFLAGS)
 
 libtotem_la_LIBADD = \
--- origsrc/totem-3.0.1/src/plugins/Makefile.plugins	2011-04-26 05:22:22.000000000 -0500
+++ src/totem-3.0.1/src/plugins/Makefile.plugins	2011-05-25 04:42:08.003931000 -0500
@@ -20,7 +20,8 @@ plugin_cflags = \
 	$(WARN_CFLAGS)
 
 plugin_ldflags = \
-	-export_dynamic \
+	-export-dynamic \
+	-no-undefined \
 	-avoid-version \
 	-module
 
