--- origsrc/totem-3.18.1/configure.ac	2015-10-12 07:14:28.000000000 -0500
+++ src/totem-3.18.1/configure.ac	2015-11-18 14:56:32.696252000 -0600
@@ -480,6 +480,42 @@ if test x$HAVE_NAUTILUS = "xyes"; then
 fi
 AM_CONDITIONAL(HAVE_NAUTILUS, test x$HAVE_NAUTILUS = "xyes")
 
+dnl ********
+dnl Caja
+dnl ********
+
+dnl Check for Caja property page build
+AC_ARG_ENABLE(caja,
+	AS_HELP_STRING([--enable-caja],[compile the caja plugin]),
+	[case "${enableval}" in
+	yes) ENABLE_CAJA=yes ;;
+	no) ENABLE_CAJA=no ;;
+	*) AC_MSG_ERROR(bad value ${enableval} for --enable-caja) ;;
+	esac],
+	[ENABLE_CAJA=yes]) dnl Default value
+
+if test x$ENABLE_CAJA = "xyes" ; then
+	PKG_CHECK_MODULES(CAJA, libcaja-extension gstreamer-tag-1.0 >= $GSTPLUG_REQS $MISSING_PLUGINS_MODULES,
+			[HAVE_CAJA=yes], [HAVE_CAJA=no])
+fi
+
+AC_ARG_WITH(cajadir,
+	    AS_HELP_STRING([--with-cajadir=DIR],[Installation path for Caja extension @<:@auto@:>@]),
+	    [ac_with_cajadir=$withval],
+	    [ac_with_cajadir=""])
+
+if test x$HAVE_CAJA = "xyes"; then
+	AC_DEFINE(HAVE_CAJA, 1, [defined if you build the Caja plugin])
+	if test "${ac_with_cajadir}" = ""; then
+		ac_with_cajadir=`pkg-config --variable=extensiondir libcaja-extension`
+	fi
+	AC_SUBST(CAJADIR)
+	AC_MSG_NOTICE([installing caja plugin in ${ac_with_cajadir}])
+	AC_SUBST([CAJADIR],[${ac_with_cajadir}])
+
+fi
+AM_CONDITIONAL(HAVE_CAJA, test x$HAVE_CAJA = "xyes")
+
 GLIB_GSETTINGS
 
 GNOME_DEBUG_CHECK
--- origsrc/totem-3.18.1/data/Makefile.am	2015-09-01 05:51:31.000000000 -0500
+++ src/totem-3.18.1/data/Makefile.am	2015-11-18 18:32:42.610221900 -0600
@@ -121,4 +121,20 @@ EXTRA_DIST += $(convert_DATA)
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = totem.pc
 
+# Caja properties page
+if HAVE_CAJA
+cajadir = $(datadir)/caja/extensions
+caja_DATA = libcaja-totem-properties-page.caja-extension
+
+%.caja-extension: %.caja-extension.in
+	$(AM_V_GEN) sed -e "s|@VERSION[@]|$(VERSION)|" \
+	$< > $@ || rm $@
+
+EXTRA_DIST += libcaja-totem-properties-page.caja-extension.in
+
+CLEANFILES += \
+	$(caja_DATA)
+endif # HAVE_CAJA
+
+
 -include $(top_srcdir)/git.mk
--- origsrc/totem-3.18.1/data/libcaja-totem-properties-page.caja-extension.in	1969-12-31 18:00:00.000000000 -0600
+++ src/totem-3.18.1/data/libcaja-totem-properties-page.caja-extension.in	2015-11-18 18:32:55.402346300 -0600
@@ -0,0 +1,7 @@
+[Caja Extension]
+Icon=totem
+Name=Totem properties
+Description=Shows details for multimedia files
+Version=@VERSION@
+Author=Andrew Sobala <aes@gnome.org>;Bastien Nocera <hadess@hadess.net>
+Copyright=Copyright (C) 2000, 2001 Eazel Inc.\nCopyright (C) 2003 Andrew Sobala <aes@gnome.org>\nCopyright (C) 2005 Bastien Nocera <hadess@hadess.net>
--- origsrc/totem-3.18.1/src/Makefile.am	2015-11-18 17:12:23.693297200 -0600
+++ src/totem-3.18.1/src/Makefile.am	2015-11-18 16:52:14.600261900 -0600
@@ -213,6 +213,35 @@ test_properties_page_LDADD = \
 	$(NAUTILUS_LIBS)
 endif # HAVE_NAUTILUS
 
+# Caja Property Page
+if HAVE_CAJA
+cajadir = $(CAJADIR)
+caja_LTLIBRARIES = libcaja-totem-properties-page.la
+
+libcaja_totem_properties_page_la_SOURCES = \
+	caja-totem-properties-main.c		\
+	caja-totem-properties-view.c		\
+	caja-totem-properties-view.h
+
+libcaja_totem_properties_page_la_CPPFLAGS = \
+	-I$(top_srcdir)/src/properties		\
+	-I$(top_builddir)/data			\
+	-DG_LOG_DOMAIN=\""TotemPropertiesPage"\"\
+	$(AM_CPPFLAGS)
+
+libcaja_totem_properties_page_la_CFLAGS =	\
+	$(CAJA_CFLAGS)		\
+	$(AM_CFLAGS)
+
+libcaja_totem_properties_page_la_LDFLAGS = \
+	-export_dynamic -avoid-version -module -no-undefined	\
+	$(AM_LDFLAGS)
+
+libcaja_totem_properties_page_la_LIBADD =			\
+	properties/libbaconvideowidgetproperties.la		\
+	$(CAJA_LIBS)
+endif # HAVE_CAJA
+
 # Totem Audio Preview for Nautilus
 totem_audio_preview_SOURCES = \
 	totem-audio-preview.c		\
