commit 6bf1e1d06d1a93b727f2fa7a4ba5c6dae2977199

--- origsrc/totem-3.26.0/docs/reference/meson.build	2017-09-11 07:30:27.000000000 -0500
+++ src/totem-3.26.0/docs/reference/meson.build	2018-02-14 14:53:58.312083100 -0600
@@ -89,7 +89,7 @@ gnome.gtkdoc(
     '--ignore-headers=' + ' '.join(private_headers),
   ],
   mkdb_args: [
-    '--sgml-mode',
+    '--xml-mode',
     '--output-format=xml',
     '--name-space=' + meson.project_name()
   ],
